packages/newspringchurchapp/src/content-single/WeekendContentItem/index.js

```diff
@@ -36,37 +41,48 @@ const WeekendContentItem = ({ content, loading }) => {
           <BackgroundView>
             <StretchyView>
               {({ Stretchy, ...scrollViewProps }) => (
-                <FlexedScrollView {...scrollViewProps}>
-                  <Header hasMedia={content.videos && content.videos.sources}>
-                    <ThemeMixin mixin={{ type: 'dark' }}>
-                      {coverImageSources.length || loading ? (
-                        <Stretchy
-                          background
-                          style={{ backgroundColor: theme.colors.primary }}
-                        >
-                          <GradientOverlayImage
-                            isLoading={!coverImageSources.length && loading}
-                            overlayColor={theme.colors.primary}
-                            overlayType="featured"
-                            source={coverImageSources}
-                          />
-                        </Stretchy>
-                      ) : null}
-                      <CardLabel
-                        title={
-                          content.parentChannel && content.parentChannel.name
-                        }
-                      />
-                      <H2 padded isLoading={!content.title && loading}>
-                        {content.title}
-                      </H2>
-                      <HTMLContent contentId={content.id} />
-                    </ThemeMixin>
-                  </Header>
-                  <MediaControls contentId={content.id} />
-                  <Features contentId={content.id} asNotes />
-                  <HorizontalContentFeed contentId={content.id} />
-                </FlexedScrollView>
+                <FlexedAvoidantView
```

Comment:
This makes the text input adjust for the keyboard. Core won't have this because they're not doing free form notes.

Reply:
Ok so then I think our signature in this file should be as small as possible, instead of adding all this here, simply replace the `ScrollView` above with a custom component containing `AvoidingView` and `ScrollView` with all your props in a different file. So something like this:

```jsx
const FlexedScrollView = styled({ flex: 1 })(CustomAvoidingScrollView);
```

Where `CustomAvoidingScrollView` comes from a different file entirely in the same directory. What do you think?